docs: add jsdocs and type definitions

snickbit opened the pull request:
I needed to write type definitions for this since my project is in TypeScript, so I figured I might as well take the time to add them directly to the package.

- Added basic JSDoc comments to all methods
- Included descriptions from the existing doc when possible
- Added typescript and tsc dev dependencies for type generation
- Ran linter & tests

snickbit committed:
docs: add jsdocs and type definitions

snickbit committed:
build: add semistandard as devDependency

kleinron commented:
Hi @snickbit!
Great work, my fellow contributor!
A few comments:
1. At least on my machine, running the linter produced 3 error messages. You can run `lint-with-fix` to automatically fix these errors and push again.
2. Please increase the version in `package.json`. Since this is a significant change, I think version `0.3.0` is a good choice.
3. In `package.json`, the `types` element is pointing to the globally installed `tsc`. Please use the local (and thus portable) version installed in `./node_modules/.bin/tsc`. Eventually the line should look like: 
`"types": "./node_modules/.bin/tsc"`

snickbit committed:
build(types): use local typescript/tsc bin

Opted for the node bin for cross-platform compatability.

snickbit committed:
chore(deps): remove deprecated tsc package

snickbit committed:
style: reapply semistandard fix

snickbit committed:
chore(bump): bump minor version to 0.3.0

snickbit commented:
I updated it with your requests.

1. My IDE auto-formatted on commit the first time, so I missed those 3 spots. Whoops!
2. Bumped the version as requested
3. I updated the "types" script, but I used `node ./node_modules/typescript/bin/tsc` to maintain cross-platform compatibility, since `./node_modules/.bin/tsc` doesn't work on Windows. I hope that's okay!

kleinron commented on types/util.d.ts:
```diff
@@ -0,0 +1,14 @@
+ * @return {void}
+ */
+export function bindMethods(): void;
+export class bindMethods {
+}
```
are you sure this is needed? maybe line #4 alone is enough?

snickbit replied:
Since you're binding methods to `this` it's considered a class constructor, so technically it's needed for TypeScript.

kleinron commented:
please also note that after resolving the linter issue, the relevant `.map` files should be regenerated and pushed, too.

snickbit committed:
chore(types): rebuild types after linting

snickbit commented:
Sorry about that. That's what I get for trying to do it quickly while at work.
